src/sail_lean_backend/Sail/Sail.lean

```diff
@@ -1,19 +1,37 @@
 namespace Sail
 namespace BitVec
 
-def length {w: Nat} (_: BitVec w): Nat := w
+def length {w : Nat} (_ : BitVec w) : Nat := w
 
-def signExtend {w: Nat} (x: BitVec w) (w': Nat) : BitVec w' :=
+def signExtend {w : Nat} (x : BitVec w) (w' : Nat) : BitVec w' :=
   x.signExtend w'
 
-def zeroExtend {w: Nat} (x: BitVec w) (w': Nat) : BitVec w' :=
+def zeroExtend {w : Nat} (x : BitVec w) (w' : Nat) : BitVec w' :=
   x.zeroExtend w'
 
-def truncate {w: Nat} (x: BitVec w) (w': Nat) : BitVec w' :=
+def truncate {w : Nat} (x : BitVec w) (w' : Nat) : BitVec w' :=
   x.truncate w'
 
-def truncateLSB {w: Nat} (x: BitVec w) (w': Nat) : BitVec w' :=
+def truncateLsb {w : Nat} (x : BitVec w) (w' : Nat) : BitVec w' :=
   x.extractLsb' 0 w'
 
+def extractLsb {w : Nat} (x : BitVec w) (hi lo : Nat) : BitVec (hi - lo + 1) :=
+  x.extractLsb hi lo
+
+def updateSubrange' {w : Nat} (x : BitVec w) (start len : Nat) (y : BitVec len) : BitVec w :=
+  let mask := ~~~(((BitVec.allOnes len).zeroExtend w) <<< start)
+  let y' := mask ||| ((y.zeroExtend w) <<< start)
+  x &&& y'
+
+def updateSubrange {w : Nat} (x : BitVec w) (hi lo : Nat) (y : BitVec (hi - lo + 1)) : BitVec w :=
+  updateSubrange' x lo _ y
+
 end BitVec
 end Sail
+
+structure registerRef (regstate regval a : Type) where
```

Comment:
```suggestion
structure RegisterRef (regstate regval a : Type) where
```

In this case, `RegisterRef` is a type(-valued function) and thus is written `UpperCamelCase`! (Sorry for not catching that before)